### PR TITLE
Add lambda function that consumes eventbus

### PR DESCRIPTION
**Pull request by svozza**
*Description of changes:*

This adds a new type of service called `event-consumer` that can be used to subscribe to events that have push onto the EventBridge bus. I've also added back the event publishing logic in the cart service that got lost for some reason.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,7 +17,6 @@ src/aws-lambda/personalize-delete-resources/models
 src/aws-lambda/personalize-pre-create-campaigns/models
 src/aws-lambda/personalize-pre-create-campaigns/data
 src/aws-lambda/pinpoint-auto-workshop/pinpoint-templates
-local/*
```
Comment (BastLeblanc) [inline .gitignore]: Why remove this? I suppose it was for a reason (I don't remember where)

File: src/aws-lambda/cart-event-consumer/stage.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (BastLeblanc) [inline src/aws-lambda/cart-event-consumer/stage.sh:1]: this file looks very generic, why is it in the "cart-event-consumer" folder = can it be in aws-lambda for more reuse?

**Review by BastLeblanc — COMMENTED** [review]
see my comments, rest is good

File: .gitignore (hunk 1 above)
Comment (svozza) [reply .gitignore]: Hmmm, I don't how that happened, I'll add it back in.

File: src/aws-lambda/cart-event-consumer/stage.sh (hunk 2 above)
Comment (svozza) [reply src/aws-lambda/cart-event-consumer/stage.sh:1]: I'm just copying what's there already, the `stage.sh` file in the root directory relies on this folder structure and those two scripts being a lambda dir so I needn't want to start hacking at that to make it more generic. I think changing this should probably done in `aws-samples/retail-demo-store` anyway and ported to here. I will raise an issue there.

Comment (BastLeblanc) [reply src/aws-lambda/cart-event-consumer/stage.sh:1]: ok raise the issue in the upstream .
but for our work for now, it's fine

**Review by BastLeblanc — APPROVED** [review]
LGTM when you'll remove the .gitignore

**Comment by svozza** [comment]
> LGTM when you'll remove the .gitignore

Fixed!

**Pull request merged**